### DISCUSSION TITLE
is mean_skill working correctly with more than one model?

### DISCUSSION THREAD

**Discussion by daniel-caichac-DHI** [DHI/modelskill#161] [Q&A]
Hi all, I got a question because I am not sure I am getting sensible results when using `mean_skill` with more than one model.
To summarize:
I have two runs (of slighlty different lengths) , **Run 000 with ~203,000 points** and **Run 023 with ~197,000 points**
if I have a single comparer object (`cc_aux`) and get the `mean_skill` I get what I am supposed to get (I checked against the dataframe with all my PointObservations)

![image](https://user-images.githubusercontent.com/97288080/213651442-5e974c92-f1bf-4a1c-87cb-47e332f194c7.png)

but If I add the `comparer object` of run 23 to the `comparer` of run 00 (with `cc.add_comparer(cc_aux)`) and then get the `mean_skill` my values change completely

![image](https://user-images.githubusercontent.com/97288080/213651626-0f8c3be8-95f8-4f88-b46c-ea0eb2e69b56.png)

Somehow the number of paired model-observations increases to **~8 million points**
@ecomodeller : Is this a bug or am I not understanding how the `mean_skill` works?

**Comment by daniel-caichac-DHI** [comment]
Furthermore, when I have a single comparer and look at a single observation (either comparer of run 00 or run 23) I get ~3000 points

![image](https://user-images.githubusercontent.com/97288080/213653407-5cc4495c-a6b4-4940-8734-4c11124626ed.png)

but when I do the same for the comparer with both runs, it all goes crazy and it creates ~114,000 points.

![image](https://user-images.githubusercontent.com/97288080/213653530-1fb1a489-41b6-421f-93bc-796843428a9a.png)

**Reply by daniel-caichac-DHI** [reply]
@jsmariegaard I think this is a bug...

**Reply by jsmariegaard** [reply]
Looks like it - could you make a small example that allows me to reproduce it?

**Comment by daniel-caichac-DHI** [comment]
@jsmariegaard 
I think that I found the issue, and is related to the `max_gap`, as is apparently not implemented with the multi-model compare.
I am uploading:
* Some observed dfs0 data
* Some modelled dfs0 data at same location
* A very very small notebook to highlight the issue
I had to zip it because otherwise I cannot upload it. A screenshot of the notebook is shown next.
![image](https://user-images.githubusercontent.com/97288080/214309863-52e1bd71-da8d-4ccf-a13e-e3811e88f267.png)

If I remove the `max_gap` keyword then I get the same number of points in the single compare and multi-model compare skill (around ~40,000) , however as explained in that branch, I need to have `max_gap` implemented otherwise I get **years** of interpolated model data when modeling independent events (such as storms only)

[Test_Data_multimodel.zip](https://github.com/DHI/fmskill/files/10490461/Test_Data_multimodel.zip)

**Comment by jsmariegaard** [comment]
I will have a look at your notebook today. 

And if I am getting you right, this is only a problem in the https://github.com/DHI/fmskill/tree/max_gap_interp branch and associated #148 ?

**Reply by daniel-caichac-DHI** [reply]
If I run it with that PR yes, I get the issue.
I just updated that branch and then the kword in `extract()` changes from `max_gap` to `max_model_gap` but the issue remains

**Comment by jsmariegaard** [comment]
The issue seems to be related to `add_comparer()`.

**Reply by daniel-caichac-DHI** [reply]
Yes

**Comment by daniel-caichac-DHI** [comment]
@ecomodeller   Can we tag this as a bug??

**Reply by daniel-caichac-DHI** [reply]
I already did now. 
the issue is the following:
* if I have 1 model I can add many observations with `add_comparer()`, no problem there
* but if I have 1 observation and many models, when I add the compares with `add_comparer()` some un-existent data is created.

See the error in the post above
https://github.com/DHI/fmskill/discussions/161#discussioncomment-4766856

**Reply by ecomodeller** [reply]
Bugs should be added to <https://github.com/DHI/fmskill/issues>

But I think it is being handled in https://github.com/DHI/fmskill/pull/172
